src/frontend/src/eth/services/send.services.ts

```diff
@@ -508,14 +511,45 @@ const approve = async ({
 			erc20HelperContractAddress
 		})
 	) {
-		return { transactionApproved: false };
+		return { transactionApproved: false, nonce };
 	}
 
+	const preApprovedAmount = await erc20ContractAllowance({
```

Comment:
Can we extract this logic in a dedicated function? That would scope the guard logic and make the code a bit more easy to understand, thinking at the future.

Reply:
I think I maybe understood: may you please have a look at the latests commits, WDYT?